src/platform/unix/mod.rs

```diff
@@ -1162,18 +1162,36 @@ impl UnixCmsg {
                 }
             },
             BlockingMode::Timeout(duration) => {
+                #[cfg(target_os = "linux")]
                 let events = libc::POLLIN | libc::POLLPRI | libc::POLLRDHUP;
+
+                // It's rather unfortunate that Rust's libc doesn't know that
+                // FreeBSD has POLLRDHUP, but in the mean time we can add
+                // the value ourselves.
+                // https://cgit.freebsd.org/src/tree/sys/sys/poll.h#n72
+                #[cfg(target_os = "freebsd")]
+                let events = libc::POLLIN | libc::POLLPRI | 0x4000;
+
```

Comment:
Could we instead add a POLLRDHUP constant that is conditionally defined, rather than the implementation code here?

Reply:
Most certainly.